### PR TITLE
optimize clear log cron and add console to clear log

### PR DESCRIPTION
**Pull request by BeardTech**

### CODE REVIEW

File: Console/ClearLog.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+use Mageplaza\Smtp\Helper\ClearLog as HelperClearLog;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class Delete
+ *
+ * @package Mageplaza\DeleteOrders\Console
```
Comment (alinalexandru) [inline Console/ClearLog.php:35]: Wrong package name.

File: Console/ClearLog.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ */
+class ClearLog extends Command
+{
+    /**
+     * @var HelperData
+     */
+    protected $_helperData;
```
Comment (alinalexandru) [inline Console/ClearLog.php:42]: Using underscore for protected variables is deprecated.

File: Helper/ClearLog.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /**
+     * Clean Email Log after X day(s)
+     *
+     * @return $this
+     */
+    public function execute()
+    {
+        if (!$this->helper->isEnabled()) {
```
Comment (alinalexandru) [inline Helper/ClearLog.php:88]: This check should be done in command console and in cron.
This helper should do the delete. On good example is that the console command could have a parameter that will overwrite the settings from database.
For example, the delete cron is disabled, but you should be capable to clean the table, if you want.

**Review by alinalexandru — CHANGES_REQUESTED** [review]

**Comment by KhietNguyen83** [comment]
Thanks for your contribution.
We have updated your issue.
Thank you very much!

**Pull request closed without merging**